### PR TITLE
Dependency Upgrade + Preparations for 1.0

### PR DESCRIPTION
**Pull request by ShaishavGandhi**
Ref #53

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -37,9 +37,9 @@ markdown_extensions:
 
 nav:
   - 'Overview': index.md
-  - '0.X API':
-      - 'rxdogtag': 0.x/rxdogtag/com.uber.rxdogtag/
-      - 'rxdogtag-autodispose': 0.x/rxdogtag-autodispose/com.uber.rxdogtag.autodispose/
+  - '1.X API':
```
Comment (ShaishavGandhi) [inline mkdocs.yml]: Not sure if it's worth keeping 0.x API reference considering that nothing has changed. Thoughts?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
     strategy:
       matrix:
         # TODO Add 13 once Gradle 6.0 is out
-        java_version: [1.8, 9, 10, 11, 12]
+        java_version: [1.8]
```
Comment (ShaishavGandhi) [inline .github/workflows/ci.yml]: Seems like jetpack benchmark barfs on Java 9 and above

https://github.com/uber/RxDogTag/pull/55/checks?check_run_id=492399969

Comment (ShaishavGandhi) [reply .github/workflows/ci.yml]: Update. Only 9 fails for JavaDateTimeException (while compiling)

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.2.2-bin.zip
```
Comment (ZacSweers) [inline gradle/wrapper/gradle-wrapper.properties]: Can we stick with all?

File: mkdocs.yml (hunk 1 above)
Comment (ZacSweers) [reply mkdocs.yml]: Yeah can remove after

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (ShaishavGandhi) [reply gradle/wrapper/gradle-wrapper.properties]: Hmm it auto changes to `bin` after i run `./gradlew wrapper`. Do you know if that's the correct command to run?

**Pull request merged**